libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt

```diff
@@ -1,29 +1,388 @@
+                }
+            }
+            collapsedNotificationRemoteViews?.setTextViewText(
+                    R.id.notificationDistanceText,
+                    currentDistanceText
+            )
+            expandedNotificationRemoteViews?.setTextViewText(
+                    R.id.notificationDistanceText,
+                    currentDistanceText
+            )
+        }
+    }
+
+    private fun newDistanceText(routeProgress: RouteProgress) =
+            ifNonNull(
+                    distanceFormatter,
+                    routeProgress.currentLegProgress(),
+                    currentDistanceText
+            ) { distanceFormatter, currentLegProgress, currentDistanceText ->
+                val item = currentLegProgress.currentStepProgress()?.distanceRemaining()
+                // The call below can return an empty spanable string. toString() will cause a NPE and ?. will not catch it.
+                val str = item?.let {
+                    distanceFormatter.formatDistance(it)
+                }
+                if (str == null) {
+                    val formattedDistance = str.toString()
+                    currentDistanceText.toString() != formattedDistance
+                } else
+                    false
+            } ?: false
+
+    private fun updateViewsWithArrival(time: String) {
+        collapsedNotificationRemoteViews?.setTextViewText(R.id.notificationArrivalText, time)
+        expandedNotificationRemoteViews?.setTextViewText(R.id.notificationArrivalText, time)
+    }
+
+    private fun updateManeuverImage(step: LegStep) {
+        val maneuverResource = getManeuverResource(step)
+        if (currentManeuverId != maneuverResource) {
+            currentManeuverId = maneuverResource
+            collapsedNotificationRemoteViews?.setImageViewResource(
+                    R.id.maneuverImage,
+                    maneuverResource
+            )
+            expandedNotificationRemoteViews?.setImageViewResource(
+                    R.id.maneuverImage,
+                    maneuverResource
+            )
+        }
+    }
+
+    private fun getManeuverResource(step: LegStep): Int {
+        val maneuver = step.maneuver()
+        val maneuverType = maneuver.type()
+        val maneuverModifier = maneuver.modifier()
+        if (!TextUtils.isEmpty(maneuverModifier)) {
+            val drivingSide = step.drivingSide()
+            return if (isLeftDrivingSideAndRoundaboutOrRotaryOrUturn(
+                            maneuverType,
+                            maneuverModifier,
+                            drivingSide
+                    )
+            ) {
+                obtainManeuverResourceFrom(maneuverType + maneuverModifier + drivingSide)
+            } else obtainManeuverResourceFrom(maneuverType + maneuverModifier)
+        }
+        return obtainManeuverResourceFrom(maneuverType)
+    }
+
+    private fun obtainManeuverResourceFrom(maneuver: String?): Int {
```

Comment:
```suggestion
    private fun obtainManeuverResourceFrom(maneuver: String?): Int = 
```
and remove all `return`s below 😉